OpenIGTLinkIF/MRML/vtkMRMLIGTLConnectorNode.cxx

```diff
@@ -1761,6 +1768,9 @@ int vtkMRMLIGTLConnectorNode::Stop()
 //---------------------------------------------------------------------------
 void vtkMRMLIGTLConnectorNode::PeriodicProcess()
 {
+  qSlicerLayoutManager* layoutManager = qSlicerApplication::application()->layoutManager();
+  layoutManager->pauseRender(true);
```

Comment:
There may be multiple layout managers and views independent of layout managers (e.g., in slicelets and in virtual reality views). To make sure rendering in these views can be paused, too, this mechanism has to be centralized.

Instead of calling qSlicerApplication::application()->layoutManager()->pauseRender(), call qSlicerApplication::application()->pauseRender() (new method) instead. This method would call layoutManager()->pauseRender() and it could invoke PauseRenderRequested event, which modules could observe (e.g., virtual reality module could observe the event and pause rendering in its special view). Same for resumeRender.

To In case of slicelets, layoutManager might be NULL, so only call pause/resume render if layoutManager is not NULL.

Reply:
Should PauseRenderRequested be invoked on qSlicerApplicationPrivate::AppLogic?

Reply:
Yes! The API is available in Slicer core now (https://github.com/Slicer/Slicer/commit/611a28085748e574d874a185b3ca78522cbf675b).